adoc/chapters/programming_interface.adoc

```diff
@@ -17640,40 +17635,500 @@ member function template.
+[code]#vec# object that corresponds to the first element of the "this" swizzle
+operation, etc.
+
+The order in which the elements are assigned is unspecified.
+Therefore, if the "this" view has any repeated elements in the swizzle, those
+elements may receive unspecified values.
+
+_Returns:_ A reference to the "this" [code]#+__swizzled_vec__+# view.
+
+a@
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+__swizzled_vec__& operator=(const DataT& rhs)
+----
+!====
+_Effects:_ Assigns the value [code]#rhs# to those elements of the underlying
+[code]#vec# object that have corresponding elements in the
+[code]#+__swizzled_vec__+# view.
+Elements in the underlying [code]#vec# object that do not have elements in the
+[code]#+__swizzled_vec__+# view are not assigned.
+
+_Returns:_ A reference to the "this" [code]#+__swizzled_vec__+# view.
+
+a@
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+__swizzled_vec__& operator=(const vec<DataT, NumElements>& rhs)
+----
+!====
+_Effects:_ Assigns elements from [code]#rhs# to elements of the [code]#vec#
+object that underlies this [code]#+__swizzled_vec__+# view.
+The first element of [code]#rhs# is assigned to the element of the underlying
+[code]#vec# object that corresponds to the first element of the swizzle
+operation, etc.
+
+The order in which the elements are assigned is unspecified.
+Therefore, if the [code]#+__swizzled_vec__+# view has any repeated elements in
+the swizzle, those elements may receive unspecified values.
+
+_Returns:_ A reference to the "this" [code]#+__swizzled_vec__+# view.
+|====
+
+[[table.functions.swizzled-vec]]
+.Hidden friend functions of the [code]#+__swizzled_vec__+# class template
+[separator="@"]
+|====
+a@
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+template</*unspecified*/>                                                                 (1)
+friend vec<DataT, NumElements> operatorOP(const __swizzled_vec__& lhs,
+                                          const __swizzled_vec__</*unspecified*/>& rhs)
+
+friend vec<DataT, NumElements> operatorOP(const vec<DataT, NumElements>& lhs,             (2)
+                                          const __swizzled_vec__& rhs);
+
+friend vec<DataT, NumElements> operatorOP(const __swizzled_vec__& lhs,                    (3)
+                                          const vec<DataT, NumElements>& rhs)
+
+friend vec<DataT, NumElements> operatorOP(const __swizzled_vec__& lhs, const DataT& rhs)  (4)
+friend vec<DataT, NumElements> operatorOP(const DataT& lhs, const __swizzled_vec__& rhs)  (5)
+----
+!====
+Where [code]#OP# is: [code]#pass:[+]#, [code]#-#, [code]#*#, [code]#/#,
+[code]#%#, [code]#&#, [code]#|#, [code]#^#, [code]#<<#, [code]#>>#.
+
+_Constraints:_ If [code]#OP# is one of the following: [code]#%#, [code]#&#,
+[code]#|#, [code]#^#, [code]#<<#, [code]#>>#; available only when: [code]#DataT
+!= float && DataT != double && DataT != half#.
+
+In addition, overload (1) is available only when the element data type of
```

Comment:
Maybe this wording could be generalized with something like:

`std::is_same_v<decltype(lhs)::element_type, decltype(rhs)::element_type> == true`

and 

`decltype(lhs)::size() == decltype(rhs)::size()`

Reply:
I think this is a general decision that we should make about the spec.  Should constraints always use C++ syntax, or can we use words to describe the constraints when that is clear and less verbose?

I think our current direction has been to use words in situations when it is less verbose.  For example, the other constraint for these functions reads:

> available only when: `DataT != float && DataT != double && DataT != half`

which is not real C++ syntax.

Reply:
That's a very good point, I think over time I have gradually shifted towards preferring C++ syntax to describe constraints since these can be very easily translated to an SFINAE condition, and therefore reduces chances of misinterpretation. I agree, we should have a discussion about this and decide what the convention should be, it's a very good point that the constrain above is not C++ syntax, so if we were to favour that, this should really use `is_same_v` to compare types.

Reply:
I'd like to keep the wording the way it is for now, using English words when the meaning seems clear and is less verbose.  I looked at the real C++ specification, and it seems like they also use English words for clarity.  For example, this is from C++ 27.10.8 "Exclusive scan":

> Mandates: All of
> * binary_op(init, init),
> * binary_op(init, *first), and
> * binary_op(*first, *first)
>
> are convertible to T

Notice how it uses the English words "are convertible to T" instead of using C++ syntax like `std::is_convertible`.

Reply:
I agree, best to leave it as it is, especially if this is typically how the C++ specification does it.